lib/explorer/backend/lazy_series.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Explorer.Backend.LazySeries do
+  @moduledoc """
+  This is an opaque implementation of a Series.
+
+  It represents an operation with its arguments.
+  """
+  alias Explorer.Series
+  alias Explorer.Backend
+
+  @behaviour Explorer.Backend.Series
+
+  defstruct op: nil, args: []
+
+  @doc false
+  def new(op, args) do
+    %__MODULE__{op: op, args: args}
+  end
+
+  @impl true
+  def eq(%Series{} = left, %Series{} = right), do: eq(left, right.data)
+
+  def eq(%Series{dtype: left_dtype, data: left_lazy}, value) do
+    data = new(:equal, [left_lazy, value])
+
+    Backend.Series.new(data, left_dtype)
+  end
+
+  # TODO: handle nested series
+  @impl true
+  def inspect(series, opts) do
+    import Inspect.Algebra
+
+    open = color("[", :list, opts)
+    close = color("]", :list, opts)
+    dtype = color("#{Series.dtype(series)}", :atom, opts)
+
+    data =
+      container_doc(
+        open,
+        series.data.args,
+        close,
+        opts,
+        &Explorer.Shared.to_string/2
+      )
+
+    concat([
+      color("LazySeries ", :atom, opts),
+      dtype,
+      line(),
+      open,
+      "???",
+      close,
+      line(),
+      color("Operation: ", :atom, opts),
+      color("#{series.data.op}", :atom, opts),
+      line(),
+      color("Args: ", :atom, opts),
+      data
+    ])
```

Comment:
My suggestion here is to convert the frames to Elixir AST, something like this:

```elixir
def to_elixir_ast(%{op: op, args: args}) do
  {op, meta, Enum.map(args, &to_elixir_ast/2)}
end

def to_elixir_ast(other) do
  other
end
```

And then you call `Macro.to_string` on it to get a visual representation!

The only question is if we are going to translate things like `equal` to `eq`, `add` to `+`, and so on (I think we should). If so, we do this:


```elixir
@to_elixir_op %{
  add: :+,
  subtract: :-,
  ...,
  equal: :==,
  ...
}

def to_elixir_ast(%{op: op, args: args}) do
  {Map.get(@to_elixir_op, op, op), meta, Enum.map(args, &to_elixir_ast/2)}
end

def to_elixir_ast(other) do
  other
end
```

Reply:
Cool! I will do that. Thanks!

Reply:
I did with `Code.quoted_to_algebra/1` in order to preserve formatting. Is this OK?